api/internal/handler/label/label_test.go

```diff
@@ -0,0 +1,312 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package label
+
+import (
+	"encoding/json"
+	"math/rand"
+	"testing"
+
+	"github.com/shiningrush/droplet"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+
+	"github.com/apisix/manager-api/internal/core/entity"
+	"github.com/apisix/manager-api/internal/core/store"
+)
+
+type testCase struct {
+	giveInput *ListInput
+	giveData  []interface{}
+	wantRet   interface{}
+}
+
+func TestPair_MarshalJSON(t *testing.T) {
+	p := Pair{Key: "test_key", Val: `test_val`}
+	content, err := json.Marshal(p)
+	assert.Nil(t, err, nil)
+	assert.Equal(t, `{"test_key":"test_val"}`, string(content))
+
+	p = Pair{Key: "test_key", Val: `test_val"`}
+	content, err = json.Marshal(p)
+	assert.Nil(t, err, nil)
+	assert.Equal(t, `{"test_key":"test_val\""}`, string(content))
```

Comment:
Can we also use `json.Unmarshal` to verify the manual constructed JSON?

Reply:
A piece of cake.